src/reconcile/rec_write.c

```diff
@@ -820,43 +819,46 @@ __rec_write(WT_SESSION_IMPL *session, WT_ITEM *buf, uint8_t *addr, size_t *addr_
     btree = S2BT(session);
     result_len = 0;
 
-    /* Checkpoint calls are different than standard calls. */
-    WT_ASSERT(session,
-      (!checkpoint && addr != NULL && addr_sizep != NULL) ||
-        (checkpoint && addr == NULL && addr_sizep == NULL));
-
-    /* In-memory databases shouldn't write pages. */
-    WT_ASSERT(session, !F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
-
-    /*
-     * We're passed a table's disk image. Decompress if necessary and verify the image. Always check
-     * the in-memory length for accuracy.
-     */
-    dsk = buf->mem;
-    if (compressed) {
-        WT_ASSERT(session, __wt_scr_alloc(session, dsk->mem_size, &ctmp));
-
-        memcpy(ctmp->mem, buf->data, WT_BLOCK_COMPRESS_SKIP);
+    if (DIAGNOSTIC_ASSERTS_ENABLED(session)) {
+        /* Checkpoint calls are different than standard calls. */
         WT_ASSERT(session,
-          btree->compressor->decompress(btree->compressor, &session->iface,
-            (uint8_t *)buf->data + WT_BLOCK_COMPRESS_SKIP, buf->size - WT_BLOCK_COMPRESS_SKIP,
-            (uint8_t *)ctmp->data + WT_BLOCK_COMPRESS_SKIP, ctmp->memsize - WT_BLOCK_COMPRESS_SKIP,
-            &result_len) == 0);
-        WT_ASSERT(session, dsk->mem_size == result_len + WT_BLOCK_COMPRESS_SKIP);
-        ctmp->size = result_len + WT_BLOCK_COMPRESS_SKIP;
+          (!checkpoint && addr != NULL && addr_sizep != NULL) ||
+            (checkpoint && addr == NULL && addr_sizep == NULL));
 
-        /* Return an error rather than assert because the test suite tests that the error hits. */
-        ret = __wt_verify_dsk(session, "[write-check]", ctmp);
+        /* In-memory databases shouldn't write pages. */
+        WT_ASSERT(session, !F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
 
-        __wt_scr_free(session, &ctmp);
-    } else {
-        WT_ASSERT(session, dsk->mem_size == buf->size);
+        /*
+         * We're passed a table's disk image. Decompress if necessary and verify the image. Always
+         * check the in-memory length for accuracy.
+         */
+        dsk = buf->mem;
+        if (compressed) {
+            WT_ASSERT(session, __wt_scr_alloc(session, dsk->mem_size, &ctmp));
+
+            memcpy(ctmp->mem, buf->data, WT_BLOCK_COMPRESS_SKIP);
+            WT_ASSERT(session,
+              btree->compressor->decompress(btree->compressor, &session->iface,
+                (uint8_t *)buf->data + WT_BLOCK_COMPRESS_SKIP, buf->size - WT_BLOCK_COMPRESS_SKIP,
+                (uint8_t *)ctmp->data + WT_BLOCK_COMPRESS_SKIP,
+                ctmp->memsize - WT_BLOCK_COMPRESS_SKIP, &result_len) == 0);
+            WT_ASSERT(session, dsk->mem_size == result_len + WT_BLOCK_COMPRESS_SKIP);
+            ctmp->size = result_len + WT_BLOCK_COMPRESS_SKIP;
+
+            /* Return an error rather than assert because the test suite tests that the error hits.
```

Comment:
nit: Now that these comments are multi-line they should become:
```
/*
 * Return an error rather than assert because the test suite tests that the error hits.
 */
```
Same deal for a few lines below this

Reply:
[fdcd618](https://github.com/wiredtiger/wiredtiger/pull/8606/commits/fdcd618b08c03d6d491e5c219153b4f52dceb695)